test-basic/unnestTest.js

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2021 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const assert = require('assert');
+const marklogic = require('../');
+const testconfig = require('../etc/test-config.js');
+const db = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+const op = marklogic.planBuilder;
+
+const TEAM_MEMBER_NAME_COLUMN = "teamMemberName";
+
+
+describe('test unnest function', function () {
+
+    before(function (done) {
+        const doc = {
+            office: [
+                { department: "Engineering", teamMembers: "Cindy,Alice,Dan" },
+                { department: "Sales", teamMembers: "Bob" },
+                { department: "Marketing", teamMembers: null }
+            ]
+        };
+
+        db.documents.write({
+            uri: '/optic/test/office1.json',
+            contentType: 'application/json',
+            content: doc
+        }).result(function() {
+            done();
+        }, function(error) {
+            done(error);
+        });
+    });
+    after((function(done){
+        db.documents.remove(['/optic/test/office1.json'])
```

Comment:
Best practice for tests is to delete data before a test starts, ensuring a clean database, and then that allows for not deleting data at the end. Which makes debugging a lot easier too, because you can see what data is in the database in case the test fails. 

Does this have to be deleted? I'm wondering if the best practice of "delete all data, except documents that are expected to always exist" can be implemented instead. I can guarantee that this makes the test less useful too - i.e. it's easy for me to explain how unnest works by running the test and then showing someone what data is in the database.